### DISCUSSION TITLE
Synology NAS, Docker

### DISCUSSION THREAD

**Discussion by jgramajo4** [philosowaffle/peloton-to-garmin#59] [Q&A] [answered]
Hello All,

Forgive my ignorance or dumb question I am still new to the self hosted game, docker, and Python. I am continuing to get this error (below) when loading the latest docker image on Portainer. If I am reading it somewhat correctly, is my issue an OS Environment issue (specifically that the .py cannot find Python 3.9)?

I am trying to run this docker image on my Synology NAS and I see that I have the latest Python package installed (Python 3.8.2). Which based on the documentation it should be supported. I have all the environmental set up as well (garmin and peloton usernames and pw along with the Debug one).

```
`Traceback (most recent call last):


  File "/opt/app/peloton-to-garmin.py", line 158, in <module>


    PelotonToGarmin.run(config)


  File "/opt/app/peloton-to-garmin.py", line 32, in run


    debug = os.environ['DEBUG']


  File "/usr/local/lib/python3.9/os.py", line 679, in __getitem__


    raise KeyError(key) from None


KeyError: 'DEBUG'`
```

I didn't think this was an issue with the code so I just submitted it as a discussion topic, hopefully that is correct.

**Comment by philosowaffle** [comment, marked as answer]
Hello!  I can give you some additional insight on that error.  I believe the issue is you are missing a section in your [config.ini](https://github.com/philosowaffle/peloton-to-garmin/blob/master/config.ini) file.

```
KeyError: 'DEBUG'`
```

This indicates that the script tried load the DEBUG configuration from the `config.ini` file but could not find it.  My best guess is you may be missing the DEBUG run parameter in your environment variables. (See the Required section [here](https://hub.docker.com/r/philo138/peloton-to-garmin))


@Octopusprime83 may be able to provide some more insight on how to configure this for docker.

**Comment by jgramajo4** [comment]
Hey @philosowaffle, I want to say thank you for creating this and @Octopusprime83 for creating the docker. I gave it the ole restart the container/start over as the Debug variable was included but it wasn't picking it up but on the 2nd go it worked flawlessly and it moved to garmin quickly.

You are all awesome and I hope one day my Python skills can reach your level so I can help. Have a great New Year and stay safe/take care!

**Comment by Octopusprime83** [comment]
@jgramajo4 Thanks! Just to clarify the variables needed for docker are detailed on the docker hub page for this image. In relation to the DEBUG variable this is something I mainly use for testing, but included incase others needed to set it to test mode for troubleshooting. It prevents the script from running on a 5min loop instead making it run only once then quitting.

**Comment by tejchicago** [comment]
FYI - I was struggling to get the latest version configured via the Docker UI on Synology.

I finally realized I was making it too difficult - I went the docker-compose route and was up and running immediately.

Outstanding work; I really appreciate the effort behind this.   See you on the leaderboard!
